src/System.Threading.Timer/tests/TimerFiringTests.cs

```diff
@@ -122,8 +120,8 @@ public void Timer_CanDisposeSelfInCallback()
             t.Dispose();
             are.Set();
         });
-        t = new Timer(tc, null, 1, -1);
-        Assert.True(are.WaitOne(50));
+        t = new Timer(tc, null, 10, -1);
+        Assert.True(are.WaitOne(250));
```

Comment:
This could be much larger, like many seconds.  This is a case where we're waiting for something to happen, and that thing must happen or the test should fail.  The only reason we want a timeout here at all is so that the test doesn't hang indefinitely if there is a bug.  This is the same case as when we've used 30 seconds in some of the other test suites, like FSW.